apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.IntegrationTests/SimplePubSubTest.cs

```diff
@@ -138,18 +143,19 @@ public SimplePubSubTest(PubsubFixture fixture)
             CancellationTokenSource watchdogCts = new CancellationTokenSource();
             Task.Run(async () =>
             {
+                var debugOutputPeriod1 = debugOutputPeriod ?? TimeSpan.FromSeconds(1);
                 int prevSentCount = -1;
                 int prevRecvCount = -1;
                 int noProgressCount = 0;
-                while (!watchdogCts.IsCancellationRequested)
+                while (true)
```

Comment:
Shouldn't there be some way to exit the loop without reporting deadlock? Or will the Task.Delay throw an exception which is then ignored, when watchdogCts.Token is cancelled by success?

Reply:
Yes, watchdogCts is cancelled on test completion, and the exception thrown by Task.Delay is ignored.